### PR TITLE
Fixed #1564395: newly created LXD container has zero network devices

### PR DESCRIPTION
**Pull request by dimitern**
tl;dr Fixes http://pad.lv/1564395 to unblock multi-NIC containers provisioning.

Machiner now always tries to call SetObservedNetworkConfig() initially, rather than only
once. This fixes the underlying issue - containers do not get multi-NIC config unless the
MA is bounced after bootstrap. Because of how the network config MA can observe
is merged with the provider's NetworkInterfaces() results, there is a contention between
spaces discovery still going (or not yet started) and trying to update the controller's
network config post-bootstrap (or post-create-model), which includes addresses coming
from subnets yet-to-be-discovered for the current model. In this case we don't store the
SubnetID of the link-layer device address, which later causes PrepareContainerInterfaceInfo()
provisioner API call to fail. This PR fixes the symptom of the issue, but later the process of
provisioning machines/containers and updating their network config will be improved
(using separate workers, rather than the machiner and provisioner).

Additionally, a new DiscoverSpacesCheckGate is added to the modelworker manifold,
but it's not utilized yet (in a follow-up).

Live tested on MAAS 1.9.0 / xenial.

(Review request: http://reviews.vapour.ws/r/4488/)

### CODE REVIEW

**Comment by dimitern** [comment]
$$merge$$
